### PR TITLE
Remove deprecated buffer constructor

### LINKED ISSUE #2671: SecretsManager fail on node10.x
Hi,

I upgraded lambda to nodejs10.x with aws-sdk (v2.457.0, v2.357.0). SecretsManager client  fail when trying to getSecretValue. This is an exception I received in cloudwatch.
```
ERROR	(node:7) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
```
I reverted back to nodejs8.10 and it works fine (aws-sdk v2.457.0). Let me know if I miss anything when upgrading to nodejs10.x.

Thanks,

### PR DESCRIPTION
**Pull request by AllanFly120**
##### Checklist

The SDK uses `new Buffer()` to construct buffer in NodeJS. This constructor is recently deprecated and will spill warning log when making any API call. This PR introduces a buffer constructor in the `util` module that will call `Buffer.from()` when the function is available, if not it will fall back to `new Buffer()`.

fix #2671 #2354

- [X] `npm run test` passes
- [X] changelog is added, `npm run add-change`
- [X] run `bundle exec rake docs:api` and inspect `doc/latest/index.html` if documentation is changed
- [X] run `npm run integration` if integration test is changed

### CODE REVIEW

**Comment by AllanFly120** [comment]
I think the codecov/patch test is complaining about the coverage of `util.js` which pollyfills NodeJS buffer API. Those branches can only be hit when running in Node < 4.5. It should be fine to ignore that test.

**Comment by AllanFly120** [comment]
@trivikr I have fixed the codecov. It's ready to be reviewed now

File: .changes/next-release/bugfix-Buffer-3b7f55e3.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "type": "bugfix",
+  "category": "Buffer",
+  "description": "remove deprecated new Buffer() constructor in NodeJS"
```
Comment (trivikr) [inline .changes/next-release/bugfix-Buffer-3b7f55e3.json]: `NodeJS` is officially called `Node.js`
```suggestion
  "description": "remove deprecated new Buffer() constructor in Node.js"
```

File: lib/util.js [hunk 2]
```diff
@@ -946,6 +969,8 @@ var util = {
   imdsDisabledEnv: 'AWS_EC2_METADATA_DISABLED'
 };
 
+process.on('warning', console.log);
```
Comment (trivikr) [inline lib/util.js]: Do we need to ship this?

Comment (AllanFly120) [reply lib/util.js]: Thank you! will remove this.

**Comment by spencerbeggs** [comment]
This PR also allows you to upgrade the node engines field all the way to 12x.

**Comment by samparnell** [comment]
@trivikr I'm still seeing buffer errors in v2.467.0 - did this change make that build?

Thx

**Comment by trivikr** [comment]
The fix is available in v2.467.0 as explained in comment https://github.com/aws/aws-sdk-js/issues/2671#issuecomment-498737229
Please open a [new issue](https://github.com/aws/aws-sdk-js/issues/new/choose) for related bugs and link to relevant comments in this thread.

**Pull request merged**